packages/stylist-core/src/style.rs

```diff
@@ -91,84 +86,84 @@ pub struct Style {
 }
 
 impl Style {
+    // The big method is monomorphic, so less code duplication and code bloat through generics
+    // and inlining
+    fn create_from_scopes_impl(class_prefix: &str, css: Scopes) -> Self {
+        let css = Arc::new(css);
+        // Creates the StyleKey, return from registry if already cached.
+        let key = StyleKey(css);
+        let reg = StyleRegistry::get_ref();
+        let mut reg = reg.lock().unwrap();
+
+        if let Some(m) = reg.get(&key) {
+            return m;
+        }
+
+        let new_style = Self {
+            inner: Arc::new(StyleContent {
+                class_name: format!("{}-{}", class_prefix, get_rand_str()),
+                ast: key.0.clone(),
+                style_str: OnceCell::new(),
+                key: Arc::new(key),
+            }),
+        };
+
+        #[cfg(target_arch = "wasm32")]
+        new_style.inner.mount().expect("Failed to mount");
```

Comment:
I'm not sure auto-mounting on creation is a good idea. For example, this would trigger when the application is started and loading, but the document is not ready yet? I'd rather let the user decide when to have the style mount/mounted/unmount.

Reply:
I actually thought about this when I was rewriting the struct. But I figured that it's ok to do that for the moment as most of the time this is the desired behaviour and I want to maintain some API compatibility with `css-in-rust`. And wasm files are usually loaded with fetch by a javascript wrapper in `<body />` and by that time document head is probably ready.

With that being said though, I did saw #11. I will address after this PR is merged.

Reply:
Fine with me!